src/Raven.Server/Documents/Replication/Incoming/IncomingReplicationHandler.cs

```diff
@@ -565,6 +565,10 @@ protected override long ExecuteCmd(DocumentsOperationContext context)
                                 var nonPersistentFlags = GetNonPersistentDocumentFlags();
                                 if (doc.Flags.Contain(DocumentFlags.Revision))
                                 {
+                                    if (nonPersistentFlags.Contain(NonPersistentDocumentFlags.FromResharding) == false && 
```

Comment:
why for resharding this check is irrelevant?

Reply:
It's too risky on resharing.
For example:
When you move revision from shard 0 to 1, and then from shard 1 to 0 when shard 1 still has a tombstone for that revision that was created on the first resharding, the revision will be deleted and lost forever.

Reply:
Done, fixed also the `restore` issue